Lib/os.py

```diff
@@ -773,7 +781,18 @@ def __ror__(self, other):
         new.update(self)
         return new
 
-def _createenviron():
+    if _exists("_create_environ"):
+        def refresh(self):
+            environ = _create_environ()
+            if name == 'nt':
+                data = {}
+                for key, value in environ.items():
+                    data[encodekey(key)] = value
+            else:
+                data = environ
+            self._data = data
```

Comment:
If `os.environb` exists, then `os.environ` and `os.environb` will be out of sync after calling `refresh()` if it's implemented by reassigning `_data`. Maybe this should instead `clear()` and `update()`.

```suggestion
            self._data.clear()
            self._data.update(data)
```

Reply:
Oh right, I added a test and used your code, thanks!